libimplantisomd5.c

```diff
@@ -31,240 +31,139 @@
 
 #include "md5.h"
 #include "libimplantisomd5.h"
+#include "utilities.h"
 
-#define APPDATA_OFFSET 883
-#define SIZE_OFFSET 84
-
-/* Length in characters of string used for fragment md5sum checking  */
-#define FRAGMENT_SUM_LENGTH 60
-/* FRAGMENT_COUNT must be an integral divisor or FRAGMENT_SUM_LENGTH */
-/* 60 => 2, 3, 4, 5, 6, 10, 12, 15, 20, or 30 */
-#define FRAGMENT_COUNT 20
-
-/* number of sectors to ignore at end of iso when computing sum */
-#define SKIPSECTORS 15
-
-#define MAX(x, y)  ((x > y) ? x : y)
-#define MIN(x, y)  ((x < y) ? x : y)
-
-/* finds primary volume descriptor and returns info from it */
-/* mediasum must be a preallocated buffer at least 33 bytes long */
-static int parsepvd(int isofd, char *mediasum, long long *isosize) {
-    unsigned char buf[2048];
-    long long offset;
-    unsigned char *p __attribute__((unused));
-
-    if (lseek(isofd, 16*2048, SEEK_SET) == -1)
-        return ((long long)-1);
-
-    offset = (16 * 2048);
-    for (;1;) {
-        if (read(isofd, buf, 2048L) == -1)
-            return ((long long)-1);
-
-        if (buf[0] == 1)
-            /* found primary volume descriptor */
-            break;
-        else if (buf[0] == 255)
-            /* hit end and didn't find primary volume descriptor */
-            return ((long long)-1);
-        offset += 2048L;
-    }
-
-    /* read out md5sum */
-#if 0
-    memcpy(mediasum, buf + APPDATA_OFFSET + 13, 32);
-    mediasum[32] = '\0';
-
-    for (p=mediasum; *p; p++)
-        if (*p != ' ')
-            break;
-
-    /* if the md5sum was all spaces, we didn't find md5sum */
-    if (!*p)
-        return -1;
-#endif
-
-    /* get isosize */
-    *isosize = (buf[SIZE_OFFSET]*0x1000000+buf[SIZE_OFFSET+1]*0x10000 +
-                buf[SIZE_OFFSET+2]*0x100 + buf[SIZE_OFFSET+3]) * 2048LL;
-
-    return offset;
-}
-
-
-static unsigned int writeAppData(unsigned char *appdata, char *valstr, unsigned int loc) {
-    if (loc + strlen(valstr) > 511) {
+static size_t writeAppData(unsigned char *const appdata, const char *const valstr, const size_t loc) {
+    if (loc + strlen(valstr) >= APPDATA_SIZE) {
         printf("Attempted to write too much appdata, exiting...\n");
         exit(-1);
```

Comment:
This exit should be removed and the callers should check for failure. libraries really shouldn't be calling exit, especially when we have a decent way to return error messages 1 layer above.